sys/include/net/sock.h

```diff
@@ -222,9 +222,39 @@ struct _sock_tl_ep {
      * over
      */
     uint16_t netif;
-    uint16_t port;          /**< transport layer port */
+    uint16_t port;          /**< transport layer port (in host byte order) */
 };
 
+/**
+ * @brief   Helper macro to initialize IPv4 address from integer statically.
+ *
+ * @param[in] a The most significant byte of the IPv4 address
+ * @param[in] b The second most significant byte of the IPv4 address
+ * @param[in] c The second least significant byte of the IPv4 address
+ * @param[in] d The least significant byte of the IPv4 address
+ *
+ * @return  A static IPv4 address array
+ */
+#if (__BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__)
+#define SOCK_INIT_IPV4(a, b, c, d) \
+    (a) | ((b) << 8) | ((c) << 16) | ((d) << 24)
+#else
+#define SOCK_INIT_IPV4(a, b, c, d) \
+    ((a) << 24) | ((b) << 16) | ((c) << 8) | (d)
+#endif
+
+/**
+ * @brief   Helper macro to initialize IPv4 address from integer dynamically.
+ *
+ * @param[in] ipv4  An IPv4 address array (needs to be of length 4 at minimum)
+ * @param[in] addr  The IPv4 address IPv4 address as 32-bit integer
+ */
+static inline void sock_init_ipv4(uint8_t *ipv4, uint32_t addr)
+{
+    const uint8_t a[] = SOCK_INIT_IPV4(addr);
```

Comment:
> Users will usually not supply IPv4 addresses as "uint32".
They either get it from other functions (e.g., dns reply), or hardcode it.
In that case

But a socket wrapper will ;-)